Editor/Models/RSI/RsiItem.cs

```diff
@@ -9,12 +10,28 @@ public class RsiItem
         public RsiItem(Rsi? rsi = null)
         {
             Rsi = rsi ?? new Rsi();
+            foreach (var rsiState in Rsi.States)
+            {
+                _rsiUpdateStates[rsiState.Name] = UpdateState.None;
+            }
         }
 
         public Rsi Rsi { get; }
 
+        private Dictionary<string, UpdateState> _rsiUpdateStates = new();
+
         public RsiSize Size => Rsi.Size;
 
+        public void MarkAsUpdated(RsiState state, UpdateState updateState)
+        {
+            _rsiUpdateStates[state.Name] = updateState;
+        }
+        
+        public UpdateState WasModified(RsiState state)
+        {
+            return _rsiUpdateStates[state.Name];
```

Comment:
This will crash if the key is not in the dictionary, for example when a state gets renamed. Change it to return a boolean and do TryGetValue + check if its UpdateState.Edit instead.